libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1129,17 +1178,56 @@ uint64_t PbftManager::getThreshold(PbftVoteTypes vote_type) const {
   }
 }
 
-size_t PbftManager::placeVote_(taraxa::blk_hash_t const &blockhash, PbftVoteTypes vote_type, uint64_t round,
-                               size_t step) {
+uint64_t PbftManager::getPbftSortitionThreshold(PbftVoteTypes vote_type, uint64_t pbft_period) const {
+  const uint64_t total_dpos_votes_count = final_chain_->dpos_eligible_total_vote_count(pbft_period);
+
+  switch (vote_type) {
+    case propose_vote_type:
+      return std::min<uint64_t>(NUMBER_OF_PROPOSERS, total_dpos_votes_count);
+    case soft_vote_type:
+    case cert_vote_type:
+    case next_vote_type:
+    default:
+      return std::min<uint64_t>(COMMITTEE_SIZE, total_dpos_votes_count);
+  }
+}
```

Comment:
Having getCurrentPbftSortitionThreshold and getPbftSortitionThreshold does not make sense as it shares 90% of code logic, this could lead to errors where fix switch case just for one of the functions

Reply:
will be solved by https://github.com/Taraxa-project/taraxa-node/issues/1896

Reply:
So please add TODO as we agreed for all things that will be solved ..

Reply:
sure, will do